elastic/apm-server
# Relocate most modelprocessors to apm-data

## Motivation/summary

Removes most of the modelprocessors from the apm-server repo and instead places them in the `apm-data` repository which can be re-used by other downstream services.

There's a slight awkwardness on the synchronization between the metrics defined in the `metrics_internal/fields/*_metrics.yml` and the upstream `modelprocessor.IsInternalMetricName`. To facilitate checking that all defined fields are set in the upstream repo, the `internal_metrics.go` generator has been repurposed as a linter which instructs the user to update the remote definition should the two go out of sync.

Since these metrics are not updated frequently, the nuisance of this process shouldn't be experienced very often.

## Checklist



~- [ ] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/main/CHANGELOG.asciidoc)~
~- [ ] Update [package changelog.yml](https://github.com/elastic/apm-server/blob/main/apmpackage/apm/changelog.yml) (only if changes to `apmpackage` have been made)~
~- [ ] Documentation has been updated~

## How to test these changes

Add a field in any of the `metrics_internal/fields/*_metrics.yml` and verify that:
- `make check-package` returns errors and instructs the user to update the appropriate repository.

## Related issues

- Depends on elastic/apm-data#19
